### PR TITLE
Implementing Tarjan algorithm in C

### PR DESCRIPTION
**Pull request by abdouskamel**
**Fixes issue:** #3015

**Changes:**
Hello !

Just implemented Tarjan algorithm in C. I also added the stack used in the implementation to the adjancency-list representation of graphs.

### CODE REVIEW

File: code/graph-algorithms/adjacency_lists_graph_representation/adjacency_lists_in_C/lgraph_stack.c [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ * @stack A pointer to a stack struct.
+ * @return Same as LG_StackPop.
+ */
+Node_ID
+LG_StackPeek(LG_Stack* stack)
+{
+  if (LG_StackEmpty(stack))
+    return (0);
+
+  return stack->nodes[stack->head];
```
Comment (arnavb) [inline code/graph-algorithms/adjacency_lists_graph_representation/adjacency_lists_in_C/lgraph_stack.c]: Parentheses

File: code/graph-algorithms/tarjan_algorithm_strongly_connected_components/tarjan_algorithm.c [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    ++(*curComp);
+  }
+
+  dc_colors[node_ID] = DC_BLACK;
+}
+
+/*
+ * For each node of the graph, stores in sConnectedComps the ID of its strongly connected
+ * component.
+ * This algorithm is the Tarjan Algorithm & executes in thêta(S + A);
```
Comment (arnavb) [inline code/graph-algorithms/tarjan_algorithm_strongly_connected_components/tarjan_algorithm.c]: Accent mark on theta?

File: code/graph-algorithms/tarjan_algorithm_strongly_connected_components/tarjan_algorithm.c [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+  free(nodesAccessNums);
+  free(nodesNums);
+  freeLG_Stack(&lg_stackTarjan);
+  free(dc_colors);
+
+  return (curComp);
+}
+
+#define S 5
```
Comment (arnavb) [inline code/graph-algorithms/tarjan_algorithm_strongly_connected_components/tarjan_algorithm.c]: S isn't really a nice name.

**Comment by abdouskamel** [comment]
Done. :)

File: code/graph-algorithms/tarjan_algorithm_strongly_connected_components/tarjan_algorithm.c (hunk 2 above)
Comment (abdouskamel) [reply code/graph-algorithms/tarjan_algorithm_strongly_connected_components/tarjan_algorithm.c]: Wrote it in french, i'll fix that.

Comment (arnavb) [reply code/graph-algorithms/tarjan_algorithm_strongly_connected_components/tarjan_algorithm.c]: D'Accord!

**Comment by arnavb** [comment]
Bon Travail!

**Comment by abdouskamel** [comment]
Merci beaucoup !

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**